sysutils/rawhide/Portfile

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           makefile 1.0
+PortGroup           legacysupport 1.0
+
+name                rawhide
+version             3.2
+
+description         (rh) find files using pretty C expressions
+long_description    (rh) find files using pretty C expressions
+homepage            https://raf.org/rawhide/
+license             GPL-3+
+platforms           darwin
+categories          sysutils
+depends_lib         port:pcre2 port:libmagic
+maintainers         raf.org:raf openmaintainer
+revision            0
+
+github.setup        raforg ${name} ${version} v
+github.tarball_from releases
+checksums           rmd160 685df5c44c4b6ef7ba27817fc1ea92949f81d388 \
+                    sha256 73d0f755ec3edb07c714255a4fb2a47b52b6225815fc39c5719b8330f94530ce \
+                    size 283346
+
+use_configure       yes
```

Comment:
this typically isn't used together with the `makefile` PG - are you using that PG for something?

Reply:
Thanks. I didn't know that. This is my first new Portfile. I used as a model, the Portfile for sysutils/daemon, one of my other programs. It has the makefile port group and use_configure, so I assumed that they went together and I used them both in this Portfile. It seemed sensible since make is needed, but I really don't know what the makefile port group does. I didn't see it mentioned in any documentation yet. It was just cargo culting.

Both packages use a bespoke ./configure script that isn't a GNU configure script. But daemon requires GNU make whereas rawhide only needs POSIX make. I don't know if that makes any difference.

I've removed the makefile port group and everything still works.

Reply:
The `makefile` portgroup exists for projects that have a Makefile, but no `./configure` scripts.

The `autoconf`/`configure` mechanisms usually do the work of determining the right compiler flags, linker flags, and environment specifics to use to set up the build, but for a project that only has a Makefile, the `makefile` portgroup sets the environment and passes the right arguments into `make` that the `./configure` normally would have set up.

Reply:
Thanks for that explanation. Perhaps the makefile port group was used for the port of my daemon program because the configure script (like this one) isn't a proper GNU configure script. But this port works without it (at least on my 10.14 laptop). My configure script will check for $CC and alter the Makefile to use it, but any $CFLAGS or $LDFLAGS need to be in the environment when make runs (which happens with legacysupport). Perhaps that's close enough to not need the makefile port group.